ingeniamotion/pdo.py

```diff
@@ -235,17 +235,25 @@ def run(self) -> None:
                 else:
                     if self._notify_receive_process_data is not None:
                         self._notify_receive_process_data()
-                    remaining_loop_time = self._refresh_rate - (time.time() - time_start)
+                    remaining_loop_time = self._refresh_rate - (time.perf_counter() - time_start)
                     if remaining_loop_time > 0:
-                        time.sleep(remaining_loop_time)
-                    iteration_duration = time.time() - time_start
+                        self.high_precision_sleep(remaining_loop_time)
+                    iteration_duration = time.perf_counter() - time_start
 
         def stop(self) -> None:
             """Stop the PDO exchange"""
             self._pd_thread_stop_event.set()
             self._net.stop_pdos()
             self.join()
 
+        @staticmethod
+        def high_precision_sleep(duration: float) -> None:
+            """Replaces the time.sleep() method in order to obtain
+            more precise sleeping times."""
+            start_time = time.perf_counter()
+            while duration - (time.perf_counter() - start_time) > 0:
```

Comment:
Are you sure that is thread-friendly? It could be a problem to motionlab.

Reply:
Yes, It may affect it. I've made some changes so we only use the high_precision_sleep when needed.